src/modules/acbs/acbs-authentication.service.ts

```diff
@@ -0,0 +1,94 @@
+import { HttpService } from '@nestjs/axios';
+import { Inject, Injectable } from '@nestjs/common';
+import { ConfigType } from '@nestjs/config';
+import { AxiosResponse } from 'axios';
+import { catchError, lastValueFrom } from 'rxjs';
+
+import AcbsConfig from '../../config/acbs.config';
+import { AcbsAuthenticationFailedException } from './acbs-authentication-failed.exception';
+import { IdpConnectResponse } from './dto/idp-connect-response.dto';
+
+@Injectable()
+export class AcbsAuthenticationService {
+  private static readonly sessionsPath = '/sessions';
+  private static readonly connectPath = '/idptoken/openid-connect';
+  private static readonly sessionIdCookieName = 'JSESSION';
+  private static readonly sessionIdCookieSeparator = ';';
+
+  constructor(
+    @Inject(AcbsConfig.KEY)
+    private readonly config: Pick<ConfigType<typeof AcbsConfig>, 'apiKey' | 'apiKeyHeaderName' | 'authentication'>,
+    private readonly httpService: HttpService,
+  ) {}
+
+  async getIdToken(): Promise<string> {
+    const sessionId = await this.createSession();
+    return this.getIdTokenForSession(sessionId);
+  }
+
+  private async createSession(): Promise<string> {
+    const createSessionResponse = await lastValueFrom(
+      this.httpService
+        .post<never>(
+          AcbsAuthenticationService.sessionsPath,
+          { loginName: this.config.authentication.loginName, password: this.config.authentication.password },
+          {
+            baseURL: this.config.authentication.baseUrl,
+            headers: {
+              'Content-Type': 'application/json',
+              [this.config.apiKeyHeaderName]: this.config.apiKey,
+            },
+          },
+        )
+        .pipe(
+          catchError((error: Error) => {
```

Comment:
Can we log it to pinojs logger as well?

Reply:
Did you mean log the original error or the wrapped error to the logger?

I've made the change assuming you meant the _original_ error, but let me know if you meant the wrapped error instead.

It might be worth another quick investigation though - I wonder if there's a config setting we can turn on to have the original error logged by `HttpService` automatically. I'll have a look on Monday

Reply:
Actually, I realised now that there is a config already enabled globally. We won't need it here - sorry! 
It can be removed in the next pr